system-tests/lib/system-tests.ts

```diff
@@ -135,6 +135,10 @@ type ExecOptions = {
    * Pass a function to assert on and/or modify the stdout before snapshotting.
    */
   onStdout?: (stdout: string) => string | void
+  /**
+   * Pass a function to yield the ChildProcess
+   */
+  onSpawn?: (cp: any) => void
```

Comment:
types are good, to be generic to Docker processes:

```suggestion
   * Pass a function to receive the spawned process as an argument.
   */
  onSpawn?: (sp: SpawnerResult) => void
```